javascript/webapp/src/hooks/specificationCurveManagement.ts

```diff
@@ -0,0 +1,211 @@
+	failedRefutationTaskIds: string[]
+	hovered: Maybe<string>
+	onMouseOver: (item: Maybe<Specification>) => void
+	outcomeOptions: IDropdownOption[]
+	vegaWindowDimensions: Dimensions
+	activeProcessing: Maybe<RunHistory>
+	data: Specification[]
+	selectedOutcome: string
+	setSelectedOutcome: Handler1<string>
+} {
+	const [selectedOutcome, setSelectedOutcome] = useState<string>('')
+	const data = useLoadSpecificationData()
+	const failedRefutationTaskIds = useFailedRefutationTaskIds(data)
+	const config = useSpecificationCurveConfig()
+	const hovered = useHoverState()
+	const onMouseOver = useOnMouseOver()
+	const definitions = useDefinitions()
+	const outcomeOptions = returnOutcomeOptions(definitions)
+	const vegaWindowDimensions = useVegaWindowDimensions()
+	const runHistory = useRunHistory()
+	const activeProcessing = useActiveProcessing(runHistory)
+
+	useEffect(() => {
+		if (!selectedOutcome.length && outcomeOptions.length >= 2) {
+			setSelectedOutcome(outcomeOptions[1]?.key as string)
+		}
+	}, [outcomeOptions, selectedOutcome, setSelectedOutcome])
+
+	return {
+		config,
+		failedRefutationTaskIds,
+		hovered,
+		onMouseOver,
+		outcomeOptions,
+		selectedOutcome,
+		vegaWindowDimensions,
+		activeProcessing,
+		data,
+		setSelectedOutcome,
+	}
+}
+
+export function useLoadSpecificationData(): Specification[] {
+	const [data, setData] = useState<Specification[]>([])
+	const defaultRun = useDefaultRun()
+	const defaultDatasetResult = useDefaultDatasetResult()
+
+	useEffect(() => {
+		if (defaultRun) {
+			if (!defaultRun.result?.length) {
+				setData([])
+			} else {
+				const result = defaultRun.result.map((x: any) => {
+					return row2spec(x)
+				}) as Specification[]
+
+				setData(buildOutcomeGroups(result))
+			}
+		} else if (!defaultRun) {
+			if (defaultDatasetResult) {
+				const f = async () => {
+					try {
+						const result = await csv(defaultDatasetResult?.url, row2spec)
+						setData(buildOutcomeGroups(result))
+					} catch (err) {
+						setData([])
+					}
+				}
+				f()
+			}
+		}
+	}, [setData, defaultRun, defaultDatasetResult])
+	return data
+}
+
+function buildOutcomeGroups(specifications: Specification[]) {
+	const grouped = groupBySpecification(specifications)
+
+	return grouped.sort(function (a, b) {
+		return a?.estimatedEffect - b?.estimatedEffect
+	})
+}
+function returnKeys(item: Specification) {
+	return [item.treatment, item.causalModel, item.estimator]
+}
+
+function returnGroupLetter(number: number) {
+	return String.fromCharCode(97 + number).toUpperCase()
+}
+
+function insertGroupsToObjects(groups: any) {
+	return Object.keys(groups).map((outcomeGroup: any, groupNumber: number) => {
+		return groups[outcomeGroup].map(
+			(specification: any, specificationNumber: number) => {
+				return {
+					...specification,
+					id: returnGroupLetter(specificationNumber) + groupNumber,
+				}
+			},
+		)
+	})
+}
+
+function groupBySpecification(array: Specification[]) {
+	const groups: any = {}
```

Comment:
I couldn't think of a better way of doing this.
Every specification is has 3 main shared things: treatment, causal model and estimator. if we have multiple outcomes, throughout them, the combination of these three things says that they're the same specification.
What I need is:
- For every different outcome, there's a letter.
- Every specification has a number.
- I form the prop id with that
For the same specification in different outcomes, they must have the same Number. So A1 and B1 are the same specification, in different outcomes.

What I'm doing now is: Creating an object with all the existing keys (treatment, causal model and estimator) and having their specifications inside of them as an array.
They I go through these keys, and inside I find their different outcomes. 
Each key becomes a number, and each outcome becomes a letter.